app/src/main/java/openfoodfacts/github/scrachx/openfood/views/category/activity/CategoryActivity.java

```diff
@@ -16,10 +18,28 @@ public static Intent getIntent(Context context) {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_category);
+        checkNetworkAndShowScreen();
+    }
+
+    private void checkNetworkAndShowScreen(){
+        ConnectivityManager cm =
+                (ConnectivityManager)this.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
```

Comment:
For code maintainability, can you use the Utils.isNetworkConnected? Just to cut down on rewriting of code :)

Reply:
Lol.. Did we both just wrote the same thing at same time ?